R/ggproto.R

```diff
@@ -27,12 +27,20 @@
 #' To explicitly call a methods in a parent, use
 #' `ggproto_parent(Parent, self)`.
 #'
+#' @section Working with ggproto classes:
+#' The ggproto objects constructed are build on top of environments, which has
+#' some ramifications. Environments do not follow the 'copy on modify' semantics
+#' one might be accustomed to in regular objects. Instead they have
+#' ['modify in place'](https://adv-r.hadley.nz/names-values.html#env-modify)
+#' semantics.
+#'
```

Comment:
I added this bit of my own volition, because I recall being a bit stumped by this when first working with ggproto objects.